src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/EndToEndTests/NabovarselEndToEnd.cs

```diff
@@ -0,0 +1,292 @@
+            }
+
+            #region Org instansiates form with message
+            string instanceAsString = JsonConvert.SerializeObject(instanceTemplate);
+            string xmlmelding = File.ReadAllText("Data/Files/melding.xml");
+
+            string boundary = "abcdefgh";
+            MultipartFormDataContent formData = new MultipartFormDataContent(boundary)
+            {
+                { new StringContent(instanceAsString, Encoding.UTF8, "application/json"), "instance" },
+                { new StringContent(xml, Encoding.UTF8, "application/xml"), "skjema" },
+                { new StringContent(xmlmelding, Encoding.UTF8, "application/xml"), "melding" }
+            };
+
+            Uri uri = new Uri("/dibk/nabovarsel/instances", UriKind.Relative);
+
+            /* TEST */
+
+            HttpClient client = SetupUtil.GetTestClient(_factory, "dibk", "nabovarsel");
+            string token = PrincipalUtil.GetOrgToken("dibk");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+            HttpResponseMessage response = await client.PostAsync(uri, formData);
+
+            response.EnsureSuccessStatusCode();
+
+            Assert.True(response.StatusCode == HttpStatusCode.Created);
+
+            Instance createdInstance = JsonConvert.DeserializeObject<Instance>(await response.Content.ReadAsStringAsync());
+
+            Assert.NotNull(createdInstance);
+            Assert.Equal(2, createdInstance.Data.Count);
+            #endregion
+
+            #region end user gets instance
+
+            // Reset token and client to end user
+            client = SetupUtil.GetTestClient(_factory, "dibk", "nabovarsel");
+            token = PrincipalUtil.GetToken(1337);
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+            string instancePath = "/dibk/nabovarsel/instances/" + createdInstance.Id;
+
+            HttpRequestMessage httpRequestMessage =
+            new HttpRequestMessage(HttpMethod.Get, instancePath);
+
+            response = await client.SendAsync(httpRequestMessage);
+            string responseContent = await response.Content.ReadAsStringAsync();
+            Instance instance = (Instance)JsonConvert.DeserializeObject(responseContent, typeof(Instance));
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("1337", instance.InstanceOwner.PartyId);
+            Assert.Equal("Task_1", instance.Process.CurrentTask.ElementId);
+            Assert.Equal(2, instance.Data.Count);
+            #endregion
+
+            #region end user gets application metadata
+
+            // Reset token and client to end user
+            client = SetupUtil.GetTestClient(_factory, "dibk", "nabovarsel");
+            token = PrincipalUtil.GetToken(1337);
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
```

Comment:
I guess there's no need to reset the token here as the previous step is also with the end user

Reply:
agree. removed